js/src/util/index.js

```diff
@@ -252,6 +234,35 @@ const execute = callback => {
   }
 }
 
+const executeAfterTransition = (callback, transitionElement, waitForTransition = true) => {
+  if (!waitForTransition) {
+    execute(callback)
+    return
+  }
+
+  const durationPadding = 5
```

Comment:
Although the comment was not present here before, 5 seems the magic number here thus it is better to put a nice comment :slightly_smiling_face:

Reply:
Not sure what to comment here.. I tried to find out why it's 5 but I have no idea.. Considering that one frame of a 60fps animation is about 16ms - 5ms seems basically nothing.

I guess the intention was to add some additional duration so the timeout will finish after the native transition event in case the browser dispatch it. That's hard to enforce tho since we're not starting any transition here (or tracking its state) but only adding some listener and a timeout that would trigger after the calculated duration no matter what.

Also `getTransitionDurationFromElement` only respects the first CSS duration. If there are multiple transitions defined that take longer `setTimeout` will finish too early. (This shouldn't be the case with our defaults.)

On a sidenote, native `transitionend` events in the test environment seem to be somewhat unreliable. No idea if this is some optimization the browser is doing because it's headless or the elements are rendered offscreen (because of the fixture) or what exactly is causing this. `setTimeout` on the other hand works reliable.

AFAICT the idea behind this emulation code was to dispatch a `transitionend` event if no other `transitionend` event had been triggered. Likely as a fallback for legacy browsers that don't support them.

Reply:
/CC @Johann-S for the opinion :heart: